packages/cli/src/util/file.ts

```diff
@@ -128,18 +128,26 @@ export async function downloadOrCopyFile(pathDest: string, urlOrPathSrc: string)
 export async function downloadFile(pathDest: string, url: string): Promise<void> {
   if (!fs.existsSync(pathDest)) {
     mkdir(path.dirname(pathDest));
-    await promisify(stream.pipeline)(got.stream(url), fs.createWriteStream(pathDest));
+    const response = await fetch(url);
```

Comment:
I don't think this is updated correctly as previous behavior was to stream the response which is important here due to the size of files we download, fetch pattern needs to be adapted accordingly

Reply:
You're right.
I've modified the streaming behavior. 
What do you think of this?

``` export async function downloadFile(pathDest: string, url: string): Promise<void> {
  if (!fs.existsSync(pathDest)) {
    mkdir(path.dirname(pathDest));
    const response = await fetch(url);
    if (!response.ok) {
      throw new Error(`Failed to download file from ${url}: ${response.status} ${response.statusText}`);
    }

    if (!response.body) {
      throw new Error("Response body is null");
    }

    const responseStream = stream.Readable.fromWeb(response.body as any);
    const fileStream = fs.createWriteStream(pathDest);
    await promisify(stream.pipeline)(responseStream, fileStream);
  }
}

Reply:
this should be tested by starting a node on a network with a large genesis state, eg. holesky or hoodi. Need to make sure performance is similar to previous

Reply:
I don't think i understand the above comment. 
Does it have anything to do with the code for the streaming behaviour?

Reply:
> I don't think i understand the above comment. Does it have anything to do with the code for the streaming behaviour?

my comment was that we should test the code but I noticed it's unused, so we can either completely remove it or test is separately, in any case, this PR should not change code behavior and this needs to be tested, not just assumed, so the code you linked here https://github.com/ChainSafe/lodestar/pull/7526#discussion_r2006211428 "looks correct" but part of the PR should be to make sure it actually is working as intended

Reply:
(you can ignore this part for now and work on easier refactoring (like moving remaining code), this is like the final thing that needs to be checked when all other changes are complete)

Reply:
Okay, I get it. 
Noted!